src/Symfony/Component/Validator/Constraints/Length.php

```diff
@@ -49,13 +59,15 @@ class Length extends Constraint
     public $min;
     public $charset = 'UTF-8';
     public $normalizer;
+    public $countUnit = self::COUNT_CODEPOINTS;
```

Comment:
I suggest adding `@var self::COUNT_*` on the `countUnit` property of the constraint, which will help static analyzers to know about the allowed values.

This should also be added as a `@param self::COUNT_*|null $countUnit` on the constructor, so that static analysis tool can report invalid usages of the constructor when using the named arguments (for instance in attributes)

Reply:
I'd had done that, but the other oprions/params didn't have such annotations. Now at least my new one has. ;)

Reply:
Most other params don't need annotations, as the native parameter type is already precise enough. We only add phpdoc `@param` when it provides *more* info that the native signature, not when it duplicates the exact same info.